### PR TITLE
feat: slack adapter

### PR DESCRIPTION
**Pull request by bsouthga**
- Adds basic slack adapter
- Adds rate limiting function / decorator, applies decorator to slack and jira api request methods.

- [x] @michelleshu 
- [x] @yangchristian

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,15 +2,15 @@
  * crosslead-adapters
  * https://github.com/CrossLead/crosslead-adapters
  *
- * Copyright (c) 2015 McChrystal Group
+ * Copyright (c) 2016 CrossLead
```
Comment (michelleshu) [inline lib/index.js]: Nice

**Comment by bsouthga** [comment]
For clarity around the decorator syntax...

```javascript

class Example {

  @wrapper(1)  
  methodToWrap() {

  }

}


function wrapper(param) {
  return function(target, property, propertyDescriptor) {
     // target === Example.prototype if prototype method, Example if static method
     // property === "methodToWrap"
     // propertyDescriptor === params used in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty


     // wrap Example.prototype.methodToWrap
     const original = target[property];
     target[property] = function(...args) {
       console.log(param);
       original.apply(this, args);
     }
  }
}
```

File: lib/clAdapters/slack/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          } else {
+            reject(err);
+          }
+        }
+      );
+    });
+  }
+
+  // null init function...
+  init() {}
```
Comment (michelleshu) [inline lib/clAdapters/slack/index.js]: Also left init() empty for JiraAdapter... I'm not sure if it should be required

File: package.json [hunk 3]
```diff
@@ -30,12 +30,13 @@
     "request-promise": "0.4.2"
   },
   "devDependencies": {
+    "babel-core": "6.14.0",
+    "babel-eslint": "6.1.2",
     "babel-plugin-add-module-exports": "0.2.1",
     "babel-plugin-transform-class-properties": "6.11.5",
+    "babel-plugin-transform-decorators-legacy": "^1.3.4",
```
Comment (yangchristian) [inline package.json]: Small nit to peg dep version

**Comment by yangchristian** [comment]
Looks good.

**Pull request merged**